### PR TITLE
APEXMALHAR-2484 Support of PartFileWriter for writing the part files

### PR DESCRIPTION
**Pull request by chaithu14**

### CODE REVIEW

**Comment by chaithu14** [comment]
@shubham-pathak22 @yogidevendra By mistakenly, I closed the PR https://github.com/apache/apex-malhar/pull/611. . Please review and merge.

**Comment by shubham-pathak22** [comment]
Looks good to me. Will merge it in a day if there are no further comments from the community

**Comment by shubham-pathak22** [comment]
@chaithu14 Please check why Jenkins check is failing.

File: library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java
Comment (ilganeli) [inline library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java]: ```FILE_CONTENTS[i]``` and ```FILE_CONTENTS[i].length``` are used repeatedly, may as well pull them out as variables

File: library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java
Comment (ilganeli) [inline library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java]: Nit: ```blockId```

File: library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java
Comment (ilganeli) [inline library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java]: If someone changes the number of blocks being tested then this substring may throw arrayIndexOutOfBounds. I think at the very least you should extract blockId to a class variable and do something like log10(blockId) if extracting the prefix in this way. A safer way would be to use a Regex to parse this out.

File: library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java
Comment (ilganeli) [inline library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java]: It's confusing to have i, j, and index here. Renaming them will make this clearer.

File: library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java
Comment (ilganeli) [inline library/src/test/java/org/apache/apex/malhar/lib/io/block/PartFileWriterTest.java]: Since this structure is used as the source of ground truth for the test it would be helpful to add a comment explaining how it's used in the test.

**Review by ilganeli — CHANGES_REQUESTED** [review]

**Comment by chaithu14** [comment]
@ilganeli Thanks for reviewing. Incorporated the review comments and updated the PR.

**Comment by chaithu14** [comment]
@ilganeli Please merge, if you are okay with the changes.

**Comment by ilganeli** [comment]
LGTM. Thanks.

**Comment by tweise** [comment]
For how to merge PRs see:  http://apex.apache.org/contributing.html

**Comment by pramodin** [comment]
@ilganeli Please mark the JIRA https://issues.apache.org/jira/browse/APEXMALHAR-2484 resolved and version 3.8.0 (upcoming). @chaithu14 @ilganeli please verify and manually close this PR,  the commit id in the tree is different from the one in the PR.

**Comment by ilganeli** [comment]
Will do. Why would the commit Id be different and how would I avoid this in future? I followed the commit guidelines step by step.




________________________________

**Comment by pramodin** [comment]
Probably happened because you may have rebased it. Ask the contributor to rebase when the PR is ready to be merged. There may be still some lag before you merge it and additional commits may have gotten into the master in which case you can merge it with an automatic merge commit. This process is not perfect but at-least it reduces the occurrences of these merges. If you see conflicts have contributor rebase and fix again.

**Pull request closed without merging**
